src/App.vue

```diff
@@ -21,14 +21,38 @@ import FullWidth from '@/components/layouts/FullWidth.vue'
 import Navbar from '@/components/Navbar.vue'
 import MobileNavbar from '@/components/MobileNavbar.vue'
 
-import { between, required, email } from 'vee-validate/dist/rules'
+import { between, required, email, dimensions, image } from 'vee-validate/dist/rules'
 import { extend } from 'vee-validate'
 
+function validateImageDimensions (value, { width, height }) {
+  return new Promise((resolve) => {
+    let image = new Image()
+    image.onerror = () => { return resolve(false) }
+    image.onload = () => { return resolve(image.width >= width && image.height >= height) }
+    image.src = URL.createObjectURL(value[0])
+  })
+}
+
+function validateImage (files) {
+  let regex = /\.(heic|heif|tiff|webp|ico|jpg|svg|jpeg|png|bmp|gif)$/i
+  if (Array.isArray(files)) {
+    return files.every((file) => { return regex.test(file.name) })
+  }
+  return regex.test(files.name)
+}
```

Comment:
This is good test case example. Try to write a test that asserts that only pointed extensions can be accepted.

Reply:
testValidateImage()

Reply:
i wrote basic test suite in 36caad3173826794654271ccde99587fbdaea179
also changed concat to + in next commit